src/Cart.js

```diff
@@ -82,15 +97,36 @@ class Cart {
     return _.reduce(this.items, function(memo, item) { return memo + item.quantity; }, 0);
   }
   clone() {
-    return new Cart(this.restaurant, this.items.slice());
+    return new Cart(this.restaurant, this.items.slice(0));
```

Comment:
The `slice` only performs a shallow clone of the array, not of any objects within it. That is to say:
```js
var x = this.items.slice();

x === this.items; // false

x[0] === this.items[0]; // true
```
I wasn't sure whether this was intended behaviour or not, so I thought I should mention it.

Also `slice()` and `slice(0)` do the same thing.

Reply:
Yep, I used `slice(0)` because I saw it somewhere else 😅

This `Cart` class does not really play well with Redux, I struggled for some changes to be detected. I think we may even remove it at some point, and store the items directly in the Redux store, but I wanted to make a quick refactoring.

Reply:
Yeah I think ideally the store would just contain data types that are supported by JSON (so no class instances). The benefits of this are: (1) the state tree itself contains no complex objects with hidden state, so it can be cloned reliably, and (2) it can be serialised to a JSON string, which helps with things like server-side-rendering (not relevant in this case).

But as you say, for now it's just more important that things work 😄.